sql/parser/expr.go

```diff
@@ -559,3 +575,32 @@ func (p *Parser) parseCastExpression() (expr.Expr, error) {
 
 	return expr.CastFunc{Expr: e, CastAs: tp}, nil
 }
+
+func (p *Parser) parseVarCharParam() (expr.Expr, error) {
+	if tok, pos, lit := p.ScanIgnoreWhitespace(); tok != scanner.LPAREN {
+		return nil, newParseError(scanner.Tokstr(tok, lit), []string{"("}, pos)
+	}
+
+	if tok, pos, lit := p.ScanIgnoreWhitespace(); tok != scanner.INTEGER {
+		return nil, newParseError(scanner.Tokstr(tok, lit), []string{"integer"}, pos)
+	}
+
+	if tok, pos, lit := p.ScanIgnoreWhitespace(); tok != scanner.RPAREN {
+		return nil, newParseError(scanner.Tokstr(tok, lit), []string{")"}, pos)
+	}
+
+	p.Unscan()
+	p.Unscan()
+
+	// For now we don't use the returned expr.
+	_, _, err := p.ParseExpr()
+	if err != nil {
+		return nil, err
+	}
+
+	if tok, pos, lit := p.ScanIgnoreWhitespace(); tok != scanner.RPAREN {
+		return nil, newParseError(scanner.Tokstr(tok, lit), []string{")"}, pos)
+	}
+
+	return nil, nil
```

Comment:
This is not necessary, isn't it?

Reply:
If we are not going to return the value inside (which is not an expression by the way but an integer), we might as well remove this function and use the 3 ifs above directly in the switch l318